src/common/functions.js

```diff
@@ -0,0 +1,117 @@
+export const s4 = () => {
+  return Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+};
+
+// generates uniqueID
+export const uniqueId = () => {
+  return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4();
+};
+
+// check singular or plural time for "ago"
+export const pluralCheck = s => {
+  if (s == 1) {
+    return ' ago';
+  } else {
+    return 's ago';
+  }
+};
+
+// function to convert timestamp to readable time
+export const timeConverter = timestamp => {
+  let t = new Date(timestamp * 1000);
+  let seconds = Math.floor((new Date() - t) / 1000);
+
+  // check this interval is in years/months/days/hours/minutes?
+  // total no of seconds in year are 31536000
+  let interval = Math.floor(seconds / 31536000);
+  if (interval > 1) {
+    return interval + ' year' + pluralCheck(interval);
+  }
+  // for months
+  interval = Math.floor(seconds / 2592000);
+  if (interval > 1) {
+    return interval + ' month' + pluralCheck(interval);
+  }
+  // for days
+  interval = Math.floor(seconds / 86400);
+  if (interval > 1) {
+    return interval + ' day' + pluralCheck(interval);
+  }
+  // for hours
+  interval = Math.floor(seconds / 3600);
+  if (interval > 1) {
+    return interval + ' hour' + pluralCheck(interval);
+  }
+  // for minutes
+  interval = Math.ceil(seconds / 60);
+  if (interval > 1) {
+    return interval + ' minute' + pluralCheck(interval);
+  }
+  // for seconds
+  return Math.floor(seconds) + ' second' + pluralCheck(seconds);
+};
+
+export const timeConverterShortened = timestamp => {
+  let t = new Date(timestamp * 1000);
+  let seconds = Math.floor((new Date() - t) / 1000);
+
+  // check this interval is in years/months/days/hours/minutes?
+  // total no of seconds in year are 31536000
+  let interval = Math.floor(seconds / 31536000);
+  if (interval > 1) {
+    return interval + ' y';
+  }
+
+  // for months
+  interval = Math.floor(seconds / 2592000);
```

Comment:
Consider changing naming of variables to reflect values and not have all operations share/depend on a single variable